### PR TITLE
Add extra checks to Throwable.readObject

### PR DESCRIPTION
**Pull request by fengxue-IS**
- class check on suppressedException
- construct new ArrayList to hold the suppressed exceptions

Signed-off-by: Jack Lu <Jack.S.Lu@ibm.com>

### CODE REVIEW

File: jcl/src/java.base/share/classes/java/lang/Throwable.java [hunk 1]
```diff
@@ -438,18 +438,36 @@ private void readObject(ObjectInputStream s)
...
+						/*[MSG "K0562", "Self-pointers not permitted in suppressedExceptions serial stream"]*/
+						throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K0562")); //$NON-NLS-1$
+					} else {
+						newList.add(t);
+					}
 				}
+				suppressedExceptions = newList;
 			}
+		} else {
+			System.out.println(suppressedExceptions.getClass().getName() + " ClassLoader: " + suppressedExceptions.getClass().getClassLoader().toString()); //$NON-NLS-1$
```
Comment (pshipton) [inline jcl/src/java.base/share/classes/java/lang/Throwable.java]: Please remove the debugging.

Comment (fengxue-IS) [reply jcl/src/java.base/share/classes/java/lang/Throwable.java]: fixed

**Comment by fengxue-IS** [comment]
marking WIP until testing passes after fixing review comments

**Comment by fengxue-IS** [comment]
@pshipton can you please take a look?

File: jcl/src/java.base/share/classes/java/lang/Throwable.java [hunk 2]
```diff
@@ -438,18 +438,39 @@ private void readObject(ObjectInputStream s)
...
-			suppressedExceptions = Collections.EMPTY_LIST;
-		} else {
-			for (Throwable t : suppressedExceptions) {
-				if (t == null) {
-					/*[MSG "K0561", "Null entries not permitted in suppressedExceptions serial stream"]*/
-					throw new NullPointerException(com.ibm.oti.util.Msg.getString("K0561")); //$NON-NLS-1$
-				} else if (t == this) {
-					/*[MSG "K0562", "Self-pointers not permitted in suppressedExceptions serial stream"]*/
-					throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K0562")); //$NON-NLS-1$
+		List<Throwable> newList = null;
```
Comment (pshipton) [inline jcl/src/java.base/share/classes/java/lang/Throwable.java]: If a non-null suppressedExceptions cannot be restored, the default value needs to be Collections.EMPTY_LIST

Comment (fengxue-IS) [reply jcl/src/java.base/share/classes/java/lang/Throwable.java]: fixed

**Comment by pshipton** [comment]
jenkins test sanity plinux jdk8,jdk11

**Pull request merged**